tests/integration_python/global/test_trampoline.py

```diff
@@ -158,3 +159,73 @@ def test_trampoline_dot_in_exe(pixi: Path, tmp_path: Path, trampoline_channel_1:
     exe_test = tmp_path / "bin" / exec_extension("exe.test")
     # The binary execute should succeed
     verify_cli_command([exe_test], stdout_contains="Success:")
+
+
+def test_trampoline_migrate_with_newer_trampoline(
+    pixi: Path, tmp_path: Path, trampoline_channel_1: str
+) -> None:
+    # this test will validate if new trampoline will migrate the older trampoline
+    env = {"PIXI_HOME": str(tmp_path)}
+
+    # create a dummy bin that will act as already installed package
+    dummy_trampoline = tmp_path / "bin" / exec_extension("dummy-trampoline")
+    os.makedirs(dummy_trampoline.parent, exist_ok=True)
+    dummy_trampoline.write_text("hello")
+
+    # now run install again, this time it should migrate the script to the new trampoline
+    verify_cli_command(
+        [
+            pixi,
+            "global",
+            "install",
+            "--channel",
+            trampoline_channel_1,
+            "dummy-trampoline",
+        ],
+        env=env,
+    )
+
+    assert dummy_trampoline.is_file()
+    assert is_binary(dummy_trampoline)
+
+    dummy_trampoline_json = tmp_path / "bin" / "trampoline_configuration" / "dummy-trampoline.json"
+
+    assert dummy_trampoline_json.is_file()
+    # run an update, it should say that everything is up to date
+    verify_cli_command(
+        [
+            pixi,
+            "global",
+            "update",
+        ],
+        env=env,
+        stderr_contains="Environment dummy-trampoline was already up-to-date",
+        stderr_excludes="Updated executable dummy-trampoline of environment dummy-trampoline",
+    )
+
+    # now change the trampoline binary , and verify that it will install newever
+    dummy_trampoline.unlink()
```

Comment:
```suggestion
```

Writing afterwards should overwrite anyway

Reply:
it indeed overwrite it, but for some reason it is not modified with pixi. Only If I remove it, write again, it is modified.